asteroid/data/musdb18_dataset.py

```diff
@@ -0,0 +1,174 @@
+from pathlib import Path
+import torch.utils.data
+import random
+import torch
+import tqdm
+import soundfile as sf
+
+
+class MUSDB18Dataset(torch.utils.data.Dataset):
+    """A dataset of that assumes audio sources to be stored
+    in track (subb)folder where each folder has a fixed number of sources.
+    For each track the users specifies a list of `source_files` and
+    a common `suffix`.
+    A linear mix is performed on the fly by summing the target and
+    the inferers up.
+
+    Due to the fact that all tracks comprise the exact same set
+    of sources, random track mixing augmentation technique
+    can be used, where sources from different tracks are mixed
+    together.
+
+    Example
+    =======
+    train/1/vocals.wav ---------------\
+    train/1/drums.wav -----------------+--> input (mix), output[target]
+    train/1/bass.wav ------------------|
+    train/1/other.wav ----------------/
+
+    """
+    def __init__(self,
+                 root,
+                 split='train',
+                 subset=None,
+                 source_files=['vocals', 'bass', 'drums', 'other'],
+                 suffix='.wav',
+                 seq_duration=None,
+                 samples_per_track=1,
+                 random_chunks=False,
+                 random_track_mix=False,
+                 source_augmentations=lambda audio: audio,
+                 sample_rate=44100):
+
+        """MUSDB18 torch.data.Dataset that samples from the MUSDB18 tracks
+        using track and excerpts with replacement.
+
+        Parameters
+        ----------
+        root : str
+            root path of MUSDB18HQ, please download from zenodo
+        suffix : str
+            specify the filename suffice, defaults to `.wav`
+        split : str
+            select dataset subfolder, defaults to ``train``.
+        subsets : list(str)
+            selects a list of track folders, defaults to `None` 
+            which loads all tracks
+        seq_duration : float
```

Comment:
We use `segment` in the other datasets? Can we change it for consistency?

Reply:
sure, will adopt